content/en/docs/tasks/administer-cluster/kms-provider.md

```diff
@@ -118,29 +144,92 @@ Then use the functions and data structures in the stub file to develop the serve
+  for logging purposes.  As expected, it is the inverse of the `EncryptRequest` call.  Plugins must verify that the
+  `key_id` is one that they understand - they must not attempt to decrypt data unless they are sure that it was
+  encrypted by them at an earlier time.
 
-  In response to procedure call Status, a compatible KMS plugin should return `v2alpha1` as `StatusResponse.Version`, "ok" as `StatusResponse.Healthz` and a keyID (KMS KEK ID) as `StatusResponse.KeyID`
+  The API server may perform thousands of `DecryptRequest` procedure calls on startup to fill its watch cache.  Thus
+  plugin implementations must perform these calls as quickly as possible, and should aim to keep each request's latency
+  at under 10 milliseconds.
+
+* Understanding `key_id` and Key Rotation
+
+  The `key_id` is the public, non-secret name of the remote KMS KEK that is currently in use.  It may be logged
+  during regular operation of the API server, and thus must not contain any private data.  Plugin implementations
+  are encouraged to use a hash to avoid leaking any data.  The KMS v2 metrics take care to hash this value before
+  exposing it via the `/metrics` endpoint.
+
+  The API server considers the `key_id` returned from the `Status` procedure call to be authoritative.  Thus, a change
+  to this value signals to the API server that the remote KEK has changed, and data encrypted with the old KEK should
+  be marked stale when a no-op write is performed (as described below).  If an `EncryptRequest` procedure call returns a
+  `key_id` that is different from `Status`, the response is thrown away and the plugin is considered unhealthy.  Thus
+  implementations must guarantee that the `key_id` returned from `Status` will be the same as the one returned by
+  `EncryptRequest`.  Furthermore, plugins must ensure that the `key_id` is stable and does not flip-flop between values
+  (i.e. during a remote KEK rotation).
+
+  Plugins must not re-use `key_id`s, even in situations where a previously used remote KEK has been reinstated.  For
+  example, if a plugin was using `key_id=A`, switched to `key_id=B`, and then went back to `key_id=A` - instead of
+  reporting `key_id=A` the plugin should report some derivative value such as `key_id=A_001` or use a new value such
+  as `key_id=C`.
+
+  Since the API server polls `Status` about every minute, `key_id` rotation is not immediate.  Furthermore, the API
+  server will coast on the last valid state for about three minutes.  Thus if a user wants to take a passive approach
+  to storage migration (i.e. by waiting), they must schedule a migration to occur at `3 + N + M` minutes after the
+  remote KEK has been rotated (`N` is how long it takes the plugin to observe the `key_id` change and `M` is the
+  desired buffer to allow config changes to be processed - a minimum `M` of five minutes is recommend).  Note that no
+  API server restart is required to perform KEK rotation.
```

Comment:
I'm afraid the above details are more for a developer rather than a kubernetes user.

Reply:
This whole section is under the `Developing a KMS plugin gRPC server` heading which is intended for developers who are developing KMS plugins.